app/src/main/java/com/abisoft/todocompose/TodoItemsRepository.kt

```diff
@@ -1,79 +1,65 @@
 package com.abisoft.todocompose
 
+import com.abisoft.todocompose.api.UpdateRequestData
+import com.abisoft.todocompose.model.TodoItemNetwork
+import com.abisoft.todocompose.model.TodoItemPost
 import com.example.myapplication.okhttp_client.TodoApiService
 import com.example.myapplication.okhttp_client.TodoDto
+import retrofit2.Response
 
+@Suppress("UNREACHABLE_CODE")
 class TodoItemsRepository(private val apiService: TodoApiService) {
 
-    // Elementlar ro'yxatini olish
-    suspend fun getTodos(token: String, revision: Int): Result<List<TodoDto>> {
-        return try {
-            val response = apiService.getTodoList("Bearer $token", revision)
-            if (response.isSuccessful) {
-                Result.success(response.body() ?: emptyList())
-            } else {
-                Result.failure(Exception("Error: ${response.code()} - ${response.message()}"))
-            }
-        } catch (e: Exception) {
-            Result.failure(e)
-        }
-    }
 
-    // Ro'yxatga yangi element qo'shish
-    suspend fun addTodo(token: String, revision: Int, todoItem: TodoDto): Result<TodoDto> {
-        return try {
-            val response = apiService.addTodo("Bearer $token", revision, todoItem)
-            if (response.isSuccessful) {
-                Result.success(response.body()!!)
-            } else {
-                Result.failure(Exception("Error: ${response.code()} - ${response.message()}"))
-            }
-        } catch (e: Exception) {
-            Result.failure(e)
+    suspend fun getTasksList(): List<TodoDto>? {
```

Comment:
Repository shouldn't return DTOs outside. Instead we should return models / entities. DTOs should be used only for making requests